### ISSUE TITLE
[expo-router] Problem with redirection to nested navigators (drawers)

### ISSUE THREAD

**Issue by szado** [expo/expo#24694]
### Minimal reproducible example

https://github.com/szado/expo-router-bug-repro

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

Redirection using <Redirect> component does not work when you tring to go to nested navigators and results with error. In my example these are two nested drawers. Entering directly work as expected, as well by < Link>.

Please run the attached repository with the reproduction code.

Error:
```
The action 'REPLACE' with payload {"name":"[firstParam]","params":{"firstParam":"first-param-value","secondParam":"second-param-value","screen":"[secondParam]","params":{"firstParam":"first-param-value","secondParam":"second-param-value"}},"source":"stack-0oW9PIGlEcdoH3gqFTQ04"} was not handled by any navigator.

Do you have a screen named '[firstParam]'?

If you're trying to navigate to a screen in a nested navigator, see https://reactnavigation.org/docs/nesting-navigators#navigating-to-a-screen-in-a-nested-navigator.

This is a development-only warning and won't be shown in production.
```

### Environment

  expo-env-info 1.0.5 environment info:               
    System:                                           
      OS: Windows 10 10.0.19045                       
    Binaries:                                         
      Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
      npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD   
    SDKs:                                             
      Android SDK:
        API Levels: 26, 28, 29
        Build Tools: 28.0.3, 29.0.3
        System Images: android-28 | Intel x86 Atom
    IDEs:
      Android Studio: Version  3.5.0.0 AI-191.8026.42.35.6010548
    npmPackages:
      expo: ~49.0.13 => 49.0.13
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.72.5 => 0.72.5
      react-native-web: ~0.19.6 => 0.19.9
    Expo Workflow: managed

**Comment by szado** [comment]
After updating to expo-router@2.0.10, the error related to the REPLACE action mentioned above no longer appears in the console. However, there is still an issue with redirection itself. Both the Redirect component and the imperative router.replace() do not seem to work with Drawer navigators.

**Comment by marklawlor** [comment]
This should be fixed in v2.0.14
